File: app/src/main/java/com/cxh/mvpart/util/SDCardUtils.java
package com.cxh.mvpart.util;

import android.os.Environment;
import android.os.StatFs;

import java.io.File;

/**
 * @author Hai (haigod7[at]gmail[dot]com)
 *         2017/3/6
 */
public class SDCardUtils {
	
	public static boolean isSDCardEnable() {
		return Environment.getExternalStorageState().equals(Environment.MEDIA_MOUNTED);
	}

	public static String getSDCardPath() {
		return Environment.getExternalStorageDirectory().getAbsolutePath() + File.separator;
	}

	public static long getSDCardAllSize() {
		if (isSDCardEnable()) {
			StatFs stat = new StatFs(getSDCardPath());
			long availableBlocks = (long) stat.getAvailableBlocks() - 4;
			long freeBlocks = stat.getAvailableBlocks();
			return freeBlocks * availableBlocks;
		}
		return 0;
	}

	public static long getFreeBytes(String filePath) {
		if (filePath.startsWith(getSDCardPath())) {
			filePath = getSDCardPath();
		} else {
			filePath = Environment.getDataDirectory().getAbsolutePath();
		}
		StatFs stat = new StatFs(filePath);
		long availableBlocks = (long) stat.getAvailableBlocks() - 4;
		return stat.getBlockSize() * availableBlocks;
	}

	public static String getRootDirectoryPath() {
		return Environment.getRootDirectory().getAbsolutePath();
	}
}

File: app/src/main/java/com/cxh/mvpart/util/SDCardUtils.kt
package com.cxh.mvpart.util

import android.os.Environment
import android.os.StatFs

import java.io.File

/**
 * @author Hai (haigod7[at]gmail[dot]com)
 *         2017/3/6
 */
object SDCardUtils {

    fun isSDCardEnable(): Boolean {
        return Environment.getExternalStorageState() == Environment.MEDIA_MOUNTED
    }

    fun getSDCardPath(): String {
        return Environment.getExternalStorageDirectory().absolutePath + File.separator
    }

    fun getSDCardAllSize(): Long {
        if (isSDCardEnable()) {
            val stat = StatFs(getSDCardPath())
            val availableBlocks = stat.availableBlocks.toLong() - 4
            val freeBlocks = stat.availableBlocks.toLong()
            return freeBlocks * availableBlocks
        }
        return 0
    }

    fun getFreeBytes(filePath: String): Long {
        var filePath = filePath
        if (filePath.startsWith(getSDCardPath())) {
            filePath = getSDCardPath()
        } else {
            filePath = Environment.getDataDirectory().absolutePath
        }
        val stat = StatFs(filePath)
        val availableBlocks = stat.availableBlocks.toLong() - 4
        return stat.blockSize * availableBlocks
    }

    fun getRootDirectoryPath(): String {
        return Environment.getRootDirectory().absolutePath
    }
}
